internal/serialization/snapshot.go

```diff
@@ -0,0 +1,53 @@
+package serialization
+
+import (
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+)
+
+const snapshotClassID = 32
+const snapshotFactoryID = -10002
+
+type Snapshot struct {
+	ID            int64
+	NumChunks     int64
+	NumBytes      int64
+	CreationTime  int64
+	JobID         int64
+	JobName       string
+	DagJsonString string
+}
+
+func (s *Snapshot) FactoryID() int32 {
+	return snapshotFactoryID
+}
+
+func (s *Snapshot) ClassID() int32 {
+	return snapshotClassID
+}
+
+type SnapshotFactory struct{}
+
+func (SnapshotFactory) Create(classID int32) serialization.IdentifiedDataSerializable {
+	if classID == snapshotClassID {
+		return &Snapshot{}
+	}
+	return nil
```

Comment:
It is better to panic here instead of returning `nil` so the actual error is known.

Reply:
https://github.com/hazelcast/hazelcast-commandline-client/pull/295/commits/ea6f747894efa10efd50b5296f03673e644117cb